state-synchronizer/src/chunk_response.rs

```diff
@@ -82,11 +82,20 @@ impl fmt::Display for GetChunkResponse {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         let txns_repr = match self.txn_list_with_proof.first_transaction_version {
             None => "empty".to_string(),
-            Some(first_ver) => format!(
-                "versions [{} - {}]",
-                first_ver,
-                first_ver - 1 + self.txn_list_with_proof.len() as u64
-            ),
+            Some(first_version) => {
+                let last_version = first_version
+                    .checked_add(self.txn_list_with_proof.len() as u64)
+                    .map_or_else(
+                        || "Version overflow on add".into(),
+                        |version| {
+                            version.checked_sub(1).map_or_else(
+                                || "Version overflow on sub".into(),
+                                |version| format!("{}", version),
+                            )
+                        },
+                    );
```

Comment:
I think this would be clearer if written as:

```rust
let last_version = first_version
  .checked_add(self.txn_list_with_proof.len() as u64) // first_version + txn.len()
  .and_then(|x| x.checked_sub(1)) // first_version + tx.len() - 1
  .ok_or("Version overflow".into())?;
```

Reply:
Done (with some modifications for string formatting 😄)